app/controllers/employments/EndEmploymentController.scala

```diff
@@ -256,239 +264,317 @@ class EndEmploymentController @Inject() (
-            )
-          )
-          val dataForCache = form.yesNoChoice match {
-            case Some(FormValuesConstants.YesValue) =>
-              mandatoryData ++ Map(EndEmploymentConstants.TelephoneNumberKey -> form.yesNoTextEntry.getOrElse(""))
-            case _ => mandatoryData ++ Map(EndEmploymentConstants.TelephoneNumberKey -> "")
-          }
-          journeyCacheService.cache(dataForCache) map { _ =>
-            Redirect(controllers.employments.routes.EndEmploymentController.endEmploymentCheckYourAnswers())
-          }
+  private def submitTelephoneCacheHandler(
+    form: YesNoTextEntryForm
+  )(implicit request: DataRequest[_]): Try[UserAnswers] =
+    form.yesNoChoice match {
+      case Some(yes) if yes == FormValuesConstants.YesValue =>
+        val questionCached = Messages(s"tai.label.${yes.toLowerCase}")
+        for {
+          question <- request.userAnswers.set(EndEmploymentTelephoneQuestionPage, questionCached)
+          number   <- question.set(EndEmploymentTelephoneNumberPage, form.yesNoTextEntry.getOrElse(""))
+        } yield {
+          journeyCacheNewRepository.set(number)
+          number
         }
-      )
-  }
-
-  def endEmploymentCheckYourAnswers: Action[AnyContent] = (authenticate andThen validatePerson).async {
-    implicit request =>
-      journeyCacheService
-        .collectedJourneyValues(
-          Seq(
-            EndEmploymentConstants.EmploymentIdKey,
-            EndEmploymentConstants.EndDateKey,
-            EndEmploymentConstants.TelephoneQuestionKey
-          ),
-          Seq(EndEmploymentConstants.TelephoneNumberKey)
+      case _ =>
+        val questionCached = Messages(
+          s"tai.label.${form.yesNoChoice.getOrElse(FormValuesConstants.NoValue).toLowerCase}"
         )
-        .map {
-          case Right((mandatoryValues, optionalSeq)) =>
-            val model = IncomeCheckYourAnswersViewModel(
-              employmentId = mandatoryValues.head.toInt,
-              preHeading = Messages("tai.endEmployment.preHeadingText"),
-              incomeSourceEnd = mandatoryValues(1),
-              contactableByPhone = mandatoryValues(2),
-              phoneNumber = optionalSeq.head,
-              backLinkUrl = controllers.employments.routes.EndEmploymentController.addTelephoneNumber().url,
-              submissionUrl = controllers.employments.routes.EndEmploymentController.confirmAndSendEndEmployment().url,
-              cancelUrl = controllers.employments.routes.EndEmploymentController.cancel(mandatoryValues.head.toInt).url
-            )
-            Ok(addIncomeCheckYourAnswers(model))
-          case Left(_) => Redirect(taxAccountSummaryRedirect)
+        for {
+          question <- request.userAnswers.set(EndEmploymentTelephoneQuestionPage, questionCached)
+          number   <- question.set(EndEmploymentTelephoneNumberPage, "")
+        } yield {
+          journeyCacheNewRepository.set(number)
```

Comment:
This returns a future and should be mapped/ flatmapped onto number

Reply:
Don't think we need the result from this, we just return number afterwards. Could maybe map the boolean out to do some checks but not sure that's needed

Reply:
If don't map future it will continue in background. Unless we want this behaviour it's better in my opinion to map the future onto the number. Can use setOrException to get rid of the Try then return a future from this method and make action async